web/packages/design/src/Alert/Alert.tsx

```diff
@@ -38,79 +43,145 @@ type Kind =
+        borderColor: theme.colors.text.disabled,
       };
+    default:
+      kind satisfies never;
   }
 };
 
-interface AlertProps extends SpaceProps, WidthProps {
+export interface AlertProps
+  extends React.ComponentPropsWithoutRef<'div'>,
+    SpaceProps,
+    WidthProps {
   kind?: Kind;
   linkColor?: string;
+  /** Additional description to be displayed below the main content. */
+  details?: React.ReactNode;
+  /** Overrides the icon specified by {@link AlertProps.kind}. */
+  icon?: React.ComponentType<IconProps>;
+  /** If specified, causes the alert to display a primary action button. */
+  primaryAction?: Action;
+  /** If specified, causes the alert to display a secondary action button. */
+  secondaryAction?: Action;
+  /** If `true`, the component displays a dismiss button that hides the alert. */
+  dismissible?: boolean;
 }
 
-const Alert = styled.div<AlertProps>`
-  display: flex;
-  align-items: center;
-  justify-content: center;
-  border-radius: ${p => p.theme.radii[1]}px;
+/** Specifies parameters of an action button. */
+interface Action {
+  content: React.ReactNode;
+  onClick: () => void;
+}
+
+interface ThemedAlertProps extends AlertProps {
+  theme: Theme;
+}
+
+/**
+ * Displays an in-page alert. Component's children are displayed as the alert
+ * title. Use the `details` attribute to display additional information. The
+ * alert may optionally contain up to 2 action buttons and a dismiss button.
+ */
+export const Alert = ({
+  kind = 'danger',
+  children,
+  details,
+  icon,
+  primaryAction,
+  secondaryAction,
+  dismissible,
+  ...otherProps
+}: AlertProps) => {
+  const alertIconSize = kind === 'neutral' ? 'large' : 'small';
+  const AlertIcon = icon || getAlertIcon(kind);
+  const showActions = !!(primaryAction || secondaryAction || dismissible);
+
+  const [dismissed, setDismissed] = useState(false);
+
+  if (dismissed) {
+    return null;
+  }
+
+  return (
+    <AlertContainer kind={kind} {...otherProps}>
+      <IconContainer kind={kind}>
+        <AlertIcon size={alertIconSize} />
+      </IconContainer>
+      <Box flex="1">
+        <Text typography="h3">{children}</Text>
+        {details}
+      </Box>
+      {showActions && (
+        <Flex ml={5} gap={2}>
+          {primaryAction && (
+            <Button
+              {...primaryButtonProps(kind)}
+              onClick={primaryAction.onClick}
+            >
+              {primaryAction.content}
+            </Button>
+          )}
+          {secondaryAction && (
+            <ButtonText onClick={secondaryAction.onClick}>
+              {secondaryAction.content}
+            </ButtonText>
+          )}
+          {dismissible && !dismissed && (
+            <ButtonIcon aria-label="Dismiss" onClick={() => setDismissed(true)}>
+              <Icon.Cross size="small" color="text.slightlyMuted" />
+            </ButtonIcon>
+          )}
+        </Flex>
+      )}
+    </AlertContainer>
+  );
+};
+
+Alert.displayName = 'Alert';
```

Comment:
I don't know. The only case I noticed when it ever made a difference is in the React dev tools in production (minimized) mode. Since I don't have a strong opinion here, and I see these used in quite a few places in the codebase, I'd rather leave it in places where it's been defined (unless you _know_ it's not needed, in which case I'll gladly remove it).

Reply:
I believe we can safely remove it https://legacy.reactjs.org/docs/react-component.html#displayname.